_ingest-pipelines/complex-conditionals.md

```diff
@@ -0,0 +1,440 @@
+        "_id": "_id",
+        "_source": {
+          "level": "critical"
+        },
+        "_ingest": {
+          "timestamp": "2025-04-23T10:39:25.468434835Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+## Multi-conditional processing
+
+The following pipeline:
+
+- Adds an `env` field set to `production` if it doesn't already exist.
+- Adds a `severity` field set to `major` if the value in the `status` field is greater than or equal to `500`.
+- Drops the document if the `env` field is set to `test` and the `message` field contains `debug`.
+
+```json
+PUT _ingest/pipeline/advanced_log_pipeline
+{
+  "processors": [
+    {
+      "set": {
+        "field": "env",
+        "value": "production",
+        "if": "!ctx.containsKey('env')"
+      }
+    },
+    {
+      "set": {
+        "field": "severity",
+        "value": "major",
+        "if": "ctx.status != null && ctx.status >= 500"
+      }
+    },
+    {
+      "drop": {
+        "if": "ctx.env == 'test' && ctx.message?.contains('debug')"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Use the following `_simulate` request to test the pipeline:
+
+```json
+POST _ingest/pipeline/advanced_log_pipeline/_simulate
+{
+  "docs": [
+    {
+      "_source": {
+        "status": 503,
+        "message": "Server unavailable"
+      }
+    },
+    {
+      "_source": {
+        "env": "test",
+        "message": "debug log output"
+      }
+    },
+    {
+      "_source": {
+        "status": 200,
+        "message": "OK"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+In the response, note that the first document has an `env: production` and `severity: major` fields added. The second document is dropped. The third document has an `env: production` field added:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "severity": "major",
+          "message": "Server unavailable",
+          "env": "production",
+          "status": 503
+        },
+        "_ingest": {
+          "timestamp": "2025-04-23T10:51:46.795026554Z"
+        }
+      }
+    },
+    null,
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "message": "OK",
+          "env": "production",
+          "status": 200
+        },
+        "_ingest": {
+          "timestamp": "2025-04-23T10:51:46.795048304Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+## Null-safe notation
+
+Use null-safe navigation notation (`?.`) to check if the field is `null`. Note that this notation can return `null` silently, therefore we recommend to first check if the returned value is `null` and then use operations like `.contains` or `==`.
+
+Not safe syntax:
+
+```
+"if": "ctx.message?.contains('debug')"
+```
+
+If the `message` field does not exist in the document, this request returns a `null_pointer_exception` with a message `Cannot invoke "Object.getClass()" because "value" is null`.
```

Comment:
```suggestion
If the `message` field does not exist in the document, this request returns a `null_pointer_exception` with the message `Cannot invoke "Object.getClass()" because "value" is null`.
```